packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts

```diff
@@ -96,6 +88,27 @@ export abstract class DeployRetrieveExecutor<
     }
   }
 
+  async setApiVersionOn(components: ComponentSet) {
+    // Check the SFDX configuration to see if there is an overridden api version.
+    // Project level local sfdx-config takes precedence over global sfdx-config at system level.
+    const userConfiguredApiVersion = await this.getUserConfiguredApiVersion();
+    
+    // If no user-configured Api Version is present, then get the version from the Org.
+    const orgApiVersion = await this.getOrgApiVersion();
```

Comment:
I don't love that we have to get this when we might have the userConfiguredApiVersion already

Reply:
Also a good callout!  I updated it and added in an assertion to one of the tests to verify that this only gets called if the user-configured value is not set.